packages/server/test/login.integration.ts

```diff
@@ -351,41 +355,155 @@ describe('Login Integration', () => {
         expect(totalUserCoursesUpdated).toEqual(3);
       });
 
-      it('handles new semester with unregistered courses', async () => {
-        const sem = await SemesterModel.findOne({
-          where: { season: 'Fall', year: 2022 },
+      describe('handles new semester with unregistered courses', () => {
+        const fallSecGroups = [
+          {
+            crns: [23456],
+            semester: '202310', // 2022 fall
+            name: 'Fundies 2 Accel',
+          },
+        ];
+
+        it('saves data for new professor', async () => {
+          expect(
+            await UserModel.findOne({
+              where: { email: 'liu.i@northeastern.edu' },
+            }),
+          ).toBeUndefined();
+
+          await supertest()
+            .post('/khoury_login')
+            .send({
+              email: 'liu.i@northeastern.edu',
+              campus: 1,
+              first_name: 'Iris',
+              last_name: 'Liu',
+              photo_url: 'sdf',
+              courses: fallSecGroups,
+            })
+            .expect(201);
+
+          const prof = await UserModel.findOne({
+            where: { email: 'liu.i@northeastern.edu' },
+            relations: ['courses'],
+          });
+
+          expect(prof.courses).toHaveLength(0); // Does not create user courses
+          const profSectionGroups = await ProfSectionGroupsModel.findOne({
+            where: { profId: prof.id },
+          });
+          expect(profSectionGroups.sectionGroups).toEqual(fallSecGroups);
         });
-        expect(sem).toBeUndefined();
 
-        await supertest()
-          .post('/khoury_login')
-          .send({
+        it('saves data for returning professor', async () => {
```

Comment:
This is a test for that bug we fixed at the start of this summer, where returning profs weren't getting their data saved